lib/linguist/generated.rb

```diff
@@ -825,5 +826,18 @@ def extract_html_meta(match)
         [key, val]
       end.select { |x| x.length == 2 }.to_h
     end
+
+    # Internal: Is this a generated SQLx query file?
+    #
+    # SQLx is a Rust SQL library which generates `**/.sqlx/query-*.json` files
+    # in offline mode (enabled by default).
+    #
+    # These are used to be able to compile a project without requiring
+    # the development database to be online.
+    #
+    # Returns true or false.
+    def generated_sqlx_query?
+      !!name.match(/^.*\.sqlx\/query-.+\.json$/)
```

Comment:
This will match `foo.sqlx/query-blah.json`. Note that the holding directory is `foo.sqlx`. Is this intentional?

Reply:
Ah nice one, nope that's not my intention!

I wanted to match that path even if it is under one or multiple level of sub-directories, I fixed that now 👍